jgrapht-core/src/main/java/org/jgrapht/generate/RandomGraphGenerator.java

```diff
@@ -359,11 +358,33 @@ public int getMaxEdgesForVertexNum(Graph<VV, EE> targetGraph)
         {
             int maxAllowedEdges;
             if (targetGraph instanceof SimpleGraph<?, ?>) {
-                maxAllowedEdges = numOfVertexes * (numOfVertexes - 1) / 2;
+                try {
```

Comment:
To keep things a bit more concise, instead of having 3 try/catch blocks doing the same thing, wrap the entire if statement into 1 try/catch